samples/bluetooth/direct_test_mode/src/fem/fem.c

```diff
@@ -155,3 +157,116 @@ uint32_t fem_radio_rx_ramp_up_delay_get(bool fast, nrf_radio_mode_t mode)
 
 	return ramp_up;
 }
+
+uint32_t fem_default_ramp_up_time_get(bool rx, nrf_radio_mode_t mode)
+{
+	bool fast_ramp_up = nrf_radio_modecnf0_ru_get(NRF_RADIO);
+
+	return rx ? fem_radio_rx_ramp_up_delay_get(fast_ramp_up, mode) :
+		    fem_radio_tx_ramp_up_delay_get(fast_ramp_up, mode);
+}
+
+int fem_tx_configure(uint32_t ramp_up_time)
+{
+	int32_t err;
+
+	fem_activate_event.event.timer.counter_period.end = ramp_up_time;
+
+	err = mpsl_fem_pa_configuration_set(&fem_activate_event, &fem_deactivate_evt);
+	if (err) {
+		printk("PA configuration set failed (err %d)\n", err);
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+int fem_rx_configure(uint32_t ramp_up_time)
+{
+	int32_t err;
+
+	fem_activate_event.event.timer.counter_period.end = ramp_up_time;
+
+	err = mpsl_fem_lna_configuration_set(&fem_activate_event, &fem_deactivate_evt);
+	if (err) {
+		printk("LNA configuration set failed (err %d)\n", err);
+		return -EFAULT;
+	}
+
+	return 0;
+}
+
+int fem_tx_gain_set(uint32_t gain)
+{
+	int32_t err;
+	mpsl_fem_gain_t fem_gain = {
+		.gain_db = 0,
```

Comment:
This will not work. Please have a look at [`mpsl_fem_gain_t` documentation](https://github.com/nrfconnect/sdk-nrfxlib/blob/main/mpsl/include/mpsl_fem_types.h#L32-L35).

It is impossible for this code to correctly fill the fields of `fem_gain` without consulting MPSL first. The intended usage of `mpsl_fem_pa_gain_set` is coupled with `mpsl_fem_tx_power_split` in the following manner:
```c
mpsl_tx_power_split_t split;
uint16_t              current_freq_mhz = ...;
int8_t                achieved_pwr     = mpsl_fem_tx_power_split(total_power, &split, current_freq_mhz);

mpsl_fem_pa_gain_set(split.fem);
```
The meaning of `private_setting` depends on the FEM used. Without a call to `mpsl_fem_tx_power_split` first, this function doesn't know how to set it correctly.

Reply:
It works as expected in our case. We want here to set output gain in arbitrary register values, and we don't care about physical gain on antenna

Reply:
@ahasztag setting `gain_db` and `private_setting` to `gain` unconditionally will cause errors when used with nRF21540, as explained in [`mpsl_fem_gain_t` documentation](https://github.com/nrfconnect/sdk-nrfxlib/blob/main/mpsl/include/mpsl_fem_types.h#L32-L35).

@KAGA164 as discussed F2F, the fact that this code works as expected is a coincidence. It will stop working when https://github.com/nrfconnect/sdk-nrf/pull/8812 is merged.

Instead of reverse-engineering the solution so that no errors occur we should focus either on using the API as intended or on extending it, if it is not flexible enough to allow for our use cases.

Reply:
Discussed F2F with @jciupis and @ahasztag. We decided to add function setting `gain_db` field to front-end module specific Tx gain default value and add a check for gain value range in this sample, The sample doc mention that gain is a specific value for front-end module which need to be set due to product specification